sd-custom/custom-operator/equals.js

```diff
@@ -0,0 +1,126 @@
+const moment = require("moment");
+const { getType, SD_CUSTOM_TYPES } = require("../type-assert");
+
+function dateEquals(l, r) {
+  const lType = getType(l);
+  const rType = getType(r);
+
+  if (lType !== rType) {
+    return false;
+  }
+
+  if (lType !== SD_CUSTOM_TYPES.date) {
+    return false;
+  }
+
+  return moment.utc(l).isSame(moment.utc(r));
+}
+
+function durationEquals(l, r) {
+  const lType = getType(l);
+  const rType = getType(r);
+
+  if (lType !== rType) {
+    return false;
+  }
+
+  if (lType !== SD_CUSTOM_TYPES.duration) {
+    return false;
+  }
+
+  return l.days === r.days;
+}
+
+function currencyEquals(l, r) {
+  const lType = getType(l);
+  const rType = getType(r);
+
+  if (lType !== rType) {
+    return false;
+  }
+
+  if (lType !== SD_CUSTOM_TYPES.currency) {
+    return false;
+  }
+
+  return l.value === r.value && l.type === r.type;
+}
+
+function phoneNumberEquals(l, r) {
+  const lType = getType(l);
+  const rType = getType(r);
+
+  if (lType !== rType) {
+    return false;
+  }
+
+  if (lType !== SD_CUSTOM_TYPES.phoneNumber) {
+    return false;
+  }
+
+  return (
+    l.number === r.number &&
+    l.code === r.code &&
+    l.country_code === r.country_code
```

Comment:
Do we always have both code and country_code? What's the difference b/w the 2?

Reply:
I took this from `isPhoneNumberEqual` in `apps/spotdraft-fe/src/app/core/utils/blank-json-compare.ts`.

I think we can safely remove the `country_code` check. From what I could see in draftkit, `country_code` is only used to render the flags in the phone number input component.

Reply:
Can you please confirm whether I should remove it @anirudhvarma12

Reply:
Yeah, lets do that.

Reply:
Made the change. Please squash and merge if everything looks fine.